### PR TITLE
Upgrade to FFMPEG 5.1.2 and pull our own FFMPEG builds from S3

### PR DESCRIPTION
**Pull request by schneems**
Previously we were hosting a `.deb` with the updates from https://github.com/heroku/heroku-buildpack-activestorage-preview/pull/28 we are now using a `tar.gz`.

We can get rid of the cache as we have to download from S3 either way. This change is gated on the full rollout of https://devcenter.heroku.com/changelog-items/2547 as #28 relies on updated system packages to be present.

It's worth noting that the common runtime and private spaces runtime update their stack images at different intervals. Due to this, merge will be delayed. 

Developers can test use this branch before merge by running:

```
heroku buildpacks:remove https://github.com/heroku/heroku-buildpack-activestorage-preview
heroku buildpacks:add -i 1 https://github.com/heroku/heroku-buildpack-activestorage-preview#schneems/update-ffmpeg-binaries
```

Once merged, you should switch back or your app will not receive updates or security fixes. You can switch back by running:

```
heroku buildpacks:remove https://github.com/heroku/heroku-buildpack-activestorage-preview#schneems/update-ffmpeg-binaries
heroku buildpacks:add -i 1 https://github.com/heroku/heroku-buildpack-activestorage-preview
```

If you're using pipelines, make sure you run commands on your staging app so binaries are properly promoted.

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -17,30 +27,12 @@ function indent() {
 
 echo '-----> Installing binary dependencies for ActiveStorage Preview'
 
-SFFMPEG_VERSION="4.2.3"
-
-# Store which ffmpeg version was installed in the cache to bust the cache if it changes
-if [ -f $CACHE_DIR/.apt/SFFMPEG ]; then
```
Comment (dzuelke) [inline bin/compile:23]: Should we keep some logic here that if `$CACHE_DIR/.apt/SFFMPEG` is there, we remove it? To purge old caches?

**Review by dzuelke — COMMENTED** [review]
IMO we should drop the `mp4` file from the fixtures; no need to have binaries in the repo. We can use e.g. https://samples.ffmpeg.org/MPEG-4/turn-on-off.mp4 directly (you can `ffprobe https://samples.ffmpeg.org/MPEG-4/turn-on-off.mp4`).

If you remove it, make sure to rewrite history to drop the commit, or the binary data will remain in history.

File: bin/compile [hunk 2]
```diff
@@ -17,30 +27,12 @@ function indent() {
...
-	CACHED_SFFMPEG=$(cat "$CACHE_DIR/.apt/SFFMPEG")
-else
-	CACHED_SFFMPEG=$SFFMPEG_VERSION
-fi
+SFFMPEG_VERSION="5.1.2"
```
Comment (dzuelke) [inline bin/compile:30]: Can we rename this to `FFMPEG_VERSION` please? :)

Comment (schneems) [reply bin/compile]: Updated, check and resolve please.

**Comment by schneems** [comment]
Please re-review. Let's hold of on giving an approval though until we're sure dogwood is fully rolled out. I added instructions for using this branch in the description.

**Pull request closed without merging**